spec/api-system-preferences-spec.js

```diff
@@ -35,6 +35,54 @@ describe('systemPreferences module', () => {
     })
   })
 
+  describe('systemPreferences.registerDefaults(defaults)', () => {
+    before(function () {
+      if (process.platform !== 'darwin') this.skip()
+    })
+
+    it('registers defaults', () => {
+      // to pass into registerDefaults
+      const defaultsDict = {
+        'one': 'ONE',
+        'two': 2,
+        'three': [1, 2, 3]
+      }
+
+      // to test type
+      const defaultsMap = [
+        { key: 'one', type: 'string', value: 'ONE' },
+        { key: 'two', value: 2, type: 'integer' },
+        { key: 'three', value: [1, 2, 3], type: 'array' }
+      ]
+
+      systemPreferences.registerDefaults(defaultsDict)
+
+      for (const def of defaultsMap) {
+        const [key, expectedValue, type] = [def.key, def.value, def.type]
+        const actualValue = systemPreferences.getUserDefault(key, type)
+        assert.deepEqual(actualValue, expectedValue)
+      }
+    })
+
+    it('throws when bad defaults are passed', () => {
+      const badDefaults1 = { 'one': null } // catches null values
+      const badDefaults2 = 1 // argument must be a dictionary
+      const badDefaults3 = null // argument can't be null
+
+      assert.throws(() => {
+        systemPreferences.registerDefaults(badDefaults1)
+      }, 'Invalid userDefault data provided')
+
+      assert.throws(() => {
+        systemPreferences.registerDefaults(badDefaults2)
+      }, 'Invalid userDefault data provided')
+
+      assert.throws(() => {
+        systemPreferences.registerDefaults(badDefaults3)
+      }, 'Invalid userDefault data provided')
+    })
```

Comment:
Maybe it would make sense to have a list of all invalid values like this?

```js
for (const badDefaults of [
    { 'one': null },  // catches null values
    1,  // argument must be a dictionary
    null,  // argument can't be null
]) {
  assert.throws(() => {
    systemPreferences.registerDefaults(badDefaults)
  }, 'Invalid userDefault data provided')
}
```